WMF Framework/LocationManager.swift

```diff
@@ -16,29 +20,225 @@ extension LocationManagerAccuracy {
 final public class LocationManager: NSObject, LocationManagerProtocol {
 
     /// Last know location
-    public var location: CLLocation? { fatalError() }
+    public var location: CLLocation? {
+        self.lastKnowLocation ?? self.locationManager.location
+    }
     /// Last know heading
-    public var heading: CLHeading? { fatalError() }
+    public var heading: CLHeading? {
+        self.lastKnowHeading ?? self.locationManager.heading
+    }
     /// Return `true` in case when monitoring location, in other case return `false`
     public private(set) var isUpdating = false
     /// Delegate for update location manager
     public var delegate: LocationManagerDelegate?
     /// Get current locationManager permission state
-    public var autorizationStatus: CLAuthorizationStatus { fatalError() }
+    public var autorizationStatus: CLAuthorizationStatus {
+        type(of: locationManager).authorizationStatus()
+    }
     // TODO: Temporary helper for support ObjC codebase, leter can call directly `autorizationStatus.authorized`
     /// Return `true` if user is aurthorized or authorized always
-    public var isAuthorized: Bool { fatalError() }
+    public var isAuthorized: Bool {
+        autorizationStatus.isAuthorized
+    }
 
     /// Start monitoring location and heading updates.
-    public func startMonitoringLocation() { }
+    public func startMonitoringLocation() {
+        if requestAuthorization()
+            || autorizationStatus == .denied
+            || autorizationStatus == .restricted
+            || isUpdating {
+            return
+        }
+
+        isUpdating = true
+
+        locationManager.startUpdatingLocation()
+        startUpdatingHeading()
+    }
 
     /// Stop monitoring location and heading updates.
-    public func stopMonitoringLocation() { }
+    public func stopMonitoringLocation() {
+        guard isUpdating else { return }
+        isUpdating = false
+        locationManager.stopUpdatingLocation()
+        stopUpdatingHeading()
+    }
 
     /// Create  new istance of locationManager with `LocationManagerAccuracy`
     /// - Parameter type: Accuracy for new instance of LocationManager
     public init(
         locationManager: CLLocationManager = .init(),
         type: LocationManagerAccuracy = .fine
-    ) { }
+    ) {
+        locationManager.activityType = .fitness
```

Comment:
Let's also add back the missing `    NSAssert([NSThread isMainThread], @"Location managers must be created on the main thread");`